lib/msal-browser/docs/acquire-token.md

```diff
@@ -89,10 +79,89 @@ var graphEndpoint = "https://graph.microsoft.com/v1.0/me";
 
 fetch(graphEndpoint, options)
     .then(resp => {
-            //do something with response
+        //do something with response
     });
 ```
 
+## MSAL token acquisition best practices
+
+The following are best practices to acquire tokens with MSAL for avoiding errors, performance hits and usability issues. Certain scenarios may provide exceptions to these.
+
+### Use a single PublicClientApplication instance
+
+Instantiate one `PublicClientApplication` per application and use the same instance throughout your app. This ensures that there is a single source of truth for what MSAL is performing at any given time (see: [MSAL events](events.md)) and eliminates the chance of distinct app objects making parallel interactive requests or potential cache conflicts, which might break apps, reduce performance or hinder user experience.
+
+### Always wait for promises to resolve
+
+All MSAL `acquireToken*` as well as `login*` APIs perform asynchronous operations and return promises. You should always wait for these promises to resolve, even if you do not need the promise payload.
+
+### Attempt silent request first, then interactive
+
+When requesting tokens, always use `acquireTokenSilent` first, falling back to interactive token acquisition if needed (e.g., when the `InteractionRequiredAuthError` is thrown).
+
+Concurrent silent requests are permitted. If two or more silent requests are made concurrently, only one would go to the network (if needed), but all would receive the response, as long as those requests are for the same request parameters (e.g. scopes).
+
+Concurrent interactive requests are **not** permitted. If two or more silent requests are made concurrently, only the first one will start an interaction, while the rest will fail with [interaction_in_progress](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/errors.md#interaction_in_progress) error.
+
+> **Avoiding *interaction_in_progress* errors**
```

Comment:
We have a whole [doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/errors.md#interaction_in_progress) dedicated to errors and how to resolve them. Can we link to that doc instead and update, if anything is missing? Keeping all the error information in one place is useful because the error messages themselves link to this doc

Reply:
good point! will move this to error doc instead then

Reply:
Done -moved most of this content to the error doc